Source/Particles/Collision/BinaryCollision/DSMC/DSMCFunc.cpp

```diff
@@ -22,6 +23,9 @@ DSMCFunc::DSMCFunc (
 {
     using namespace amrex::literals;
 
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE( !(std::is_same<amrex::ParticleReal, float>::value),
```

Comment:
Instead of checking that `ParticleReal` precision is not single, I'd say it is better to check that is is `double`.
```suggestion
    WARPX_ALWAYS_ASSERT_WITH_MESSAGE( (std::is_same<amrex::ParticleReal, double>::value),
```

Reply:
Agreed.

Technically even `double` or `long double`, but we do not expose the latter in build systems yet and using it would be really slow.
https://en.cppreference.com/w/cpp/language/types